perceptia-one/perceptia/src/components/QuestionView.js

```diff
@@ -101,48 +163,62 @@ class QuestionView extends React.Component {
 
     renderLogic() {
 
-        console.log(this.state.completed)
-
         if (this.state.completed) {
 
             return (
-                <Results />
+                <Results questions={this.state.questions} grades={this.state.grades}/>
             )
 
         } else {
 
+            console.log(this.state.mode==="study", this.state.paragraph_index <= this.state.counter, this.state.paragraph_id < this.state.paragraphs[this.state.paragraph_index].id)
+            if(this.state.mode==="study" && this.state.paragraph_index <= this.state.counter){
```

Comment:
based on this logic, even if study mode was selected, it would return question component in the case the other check against the counter failed. Was this intentional? Also, there are a lot of reads of values that may be null, without checks for the containing state object being null. (Such as getting a value from the paragraphs array).

Reply:
I was previously trying to use all three of those conditions, but I only wound up using the first two, it is intentional (it alternates between study and question, similar to how Emily mocked it up. I thought I had initialized all of them all the time, but I'll go back and check.